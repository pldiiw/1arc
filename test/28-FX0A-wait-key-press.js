'use strict';

// No key pressed? Decrement program counter in order to re-read the
// waitKeyPress instruction.
// Key pressed? Apply normal behaviour: save key value into register.

let engine = require('engine').initialize();
const waitKeyPress = require('instructions').waitKeyPress;

engine.data[0xB] = 230;

engine.pc = 4000;
const engine1 = waitKeyPress(engine, 0xB);

engine.keypad[7] = 1;
const engine2 = waitKeyPress(engine, 0xB);

console.log(engine1.data[0xB]);
console.log(engine1.pc);
console.log(engine2.data[0xB]);
console.log(engine2.pc);
