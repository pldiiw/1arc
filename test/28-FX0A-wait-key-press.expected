230
3098
7
4000
